### DISCUSSION TITLE
Reloading Config After Changes

### DISCUSSION THREAD

**Discussion by TequilaMockingbird** [blakeblackshear/frigate#2902] [Ask A Question] [answered]
Is it possible to have Frigate reload the config file after making changes? It doesn't seem to reload the config file when restarting Frigate from the browser UI, so perhaps I need to run docker-compose down and docker-compose up in order to see the changes?

**Comment by trythatagain** [comment]
I've noticed the same thing, when trying to restart from the GUI, more often than not it doesn't seem to reload the changes. I've started to restart container via portainer now, just to make sure.

**Comment by blakeblackshear** [comment, marked as answer]
Restarting from the GUI should cause the container to completely exit and docker should restart it. See what happens if you try restarting from the GUI twice.

**Reply by NickM-27** [reply]
Probably just an Unraid thing but when I restart from the frigate GUI (on Unraid docker) it doesn't restart just stops until I start in Unraid GUI

**Reply by TequilaMockingbird** [reply]
Restarting twice seemed to do the trick. I'm not sure this makes any sense but, I tried ten times making config changes and restarts this morning, and the first two times I needed to restart twice to invoke the new config, but after that (config changes 3 through 10) the change is invoked with a single restart. 
Although I don't understand why this would occur, it seems all good now. Many thanks Blake and huge kudos to you for your creation of Frigate. Really really good job.

**Reply by TequilaMockingbird** [reply]
Update: I have three cameras and have been editing the motion masks for each camera one by one. Frigate seems to require 2 restarts for these edits, but one restart does the trick for the edits I was doing earlier today: changing an object list item from a mouse to an elephant - not that I have elephants wandering around my patio - I was just playing. ; )

**Reply by trythatagain** [reply]
You're probably correct, I did restart it a few times more than once, but didn't twig that it needed multiple restarts, found it easier and reliable to just use Portainer to restart container. Not a biggy once you know what to do.

**Reply by blakeblackshear** [reply]
It's likely a bug causing it to only work intermittently.

**Reply by si458** [reply]
ill put my 2 pence worth in too if u may, im seeing this same issue, i have to restart the frigate twice to get it to reload any config at all?
e.g i setup a single camera for testing, then decided to add another camera in, but after first restart it never added the second camera? then after a second restart the second camera appeared?
